wgpu-hal/src/dynamic/queue.rs

```diff
@@ -0,0 +1,54 @@
+use crate::{
+    DeviceError, DynCommandBuffer, DynFence, DynResource, DynSurface, DynSurfaceTexture,
+    FenceValue, Queue, SurfaceError,
+};
+
+use super::DynResourceExt as _;
+
+pub trait DynQueue: DynResource {
```

Comment:
issue: We don't document a safety contract or adherence thereto for these methods. We should do this.